packages/syft/src/syft/core/node/new/document_store.py

```diff
@@ -294,9 +312,14 @@ def __init__(
         self.store_config = store_config
         self.init_store()
 
-    def init_store(self) -> None:
-        self.unique_cks = self.settings.unique_keys.all
-        self.searchable_cks = self.settings.searchable_keys.all
+    def init_store(self) -> Result[Ok, Err]:
+        try:
+            self.unique_cks = self.settings.unique_keys.all
+            self.searchable_cks = self.settings.searchable_keys.all
+        except BaseException as e:
```

Comment:
I think in a seperate task we can write some code which can automatically run a function, capture any exceptions and propagate them back as an Err (with some information about where it was caught) or Ok if everything is fine so we can get some clarity on where exceptions occur.